src/fragmentarium/ui/fragment/Download.tsx

```diff
@@ -59,12 +67,23 @@ export default function DownloadFragment({
     )
     setAtf(atfUrl)
 
+    let photoUrl = ''
+    if (fragment.hasPhoto && !photo) {
+      Promise.resolve(fragmentService.findPhoto(fragment)).then((photo) => {
+        photoUrl = URL.createObjectURL(photo)
+        setPhoto(photoUrl)
```

Comment:
I think if the condition is true, `setPhoto(photoUrl)` is called twice – maybe put the second call inside an `else` block?